src/clj/puppetlabs/http/client/async.clj

```diff
@@ -192,16 +191,18 @@
   [opts http-response]
   (let [headers       (get-resp-headers http-response)
         orig-encoding (headers "content-encoding")]
-    {:opts                  opts
+    {:opts opts
      :orig-content-encoding orig-encoding
-     :status                (.. http-response getStatusLine getStatusCode)
-     :headers               headers
-     :content-type          (parse-content-type (headers "content-type"))
-     :body                  (do
-		              (if (:decompress-body opts)
-		                (.process (ResponseContentEncoding.) http-response nil))
-		              (when-let [entity (.getEntity http-response)]
-		                (.getContent entity)))}))
+     :status (.. http-response getStatusLine getStatusCode)
+     :headers headers
+     :content-type (parse-content-type (headers "content-type"))
+     :body (do
+             (if (:decompress-body opts)
+               (.process (ResponseContentEncoding.)
+                         http-response
+                         (BasicHttpContext.)))
```

Comment:
This ```BasicHttpContext``` fixes an NPE in ```ResponseContentEncoding``` which previously could cope with nil but now can't, right? Is there any danger we could break something by using a different context to the request itself? 

My guess is probably not, but we could in principle create the context at the point we create the request and pass it through to here. Right now our request's context is created inside ```CloseableHttpAsyncClient``` so we'd need to make that explicit. 

I'm inclined to feel we shouldn't do this - it makes the flow on control tougher to follow and fixes no actual problem.